lib/multipass_ex.ex

```diff
@@ -1,79 +1,36 @@
 defmodule MultipassEx do
   @moduledoc """
-  Mimicking ruby multipass default encryption of aes_cbc128
-  """
-  @aes128_block_size 16
-  @encryption_bits 128
-  # this is from the ruby default OpenSSL IV and is insecure
-  @iv String.duplicate(<<0>>, 16)
+  Support for consuming and generating Multipass payloads based on the Rails
+  gem `Multipass`.
 
-  @doc """
-  Encodes a map into a multipass with a given site_key and api_key
-  """
-  @spec encode(map(), String.t(), String.t()) :: {:ok, String.t()} | {:error, any()}
-  def encode(data, site_key, api_key)
-      when is_map(data) and is_binary(site_key) and is_binary(api_key) do
-    try do
-      formatted_data =
-        data
-        |> Jason.encode!()
-        |> pad(@aes128_block_size)
+  If integrating with a service using the Rails gem care must be taken to ensure
+  `site_key` and `api_key` match between the services.
 
-      encoded_data =
-        :aes_cbc128
-        |> :crypto.block_encrypt(generate_key(site_key, api_key), @iv, formatted_data)
-        |> Base.url_encode64()
+  ## Example
 
-      {:ok, encoded_data}
-    rescue
-      err -> {:error, err}
-    end
-  end
+      iex> MultipassEx.encode(%{example: "data"}, "QTj#7I1cz7O1T$&1E0W!uus1c112##@z", "9B%k*5HOO27&#myca3MYvrpQzKRurgu*")
+      {:ok, "7tAxfhTv4nqhkywRcyBMkSLp4c16VQ2gLf7V3590k3Q"}
 
-  @doc """
-  Decodes a given multipass into a map with a given site_key and api_key
+      iex> MultipassEx.decode("7tAxfhTv4nqhkywRcyBMkSLp4c16VQ2gLf7V3590k3Q", "QTj#7I1cz7O1T$&1E0W!uus1c112##@z", "9B%k*5HOO27&#myca3MYvrpQzKRurgu*")
+      {:ok, %{"example" => "data"}}
   """
-  @spec decode(String.t(), String.t(), String.t()) :: {:ok, map()} | {:error, String.t()} | {:error, any()}
-  def decode(data, site_key, api_key)
-      when is_binary(data) and is_binary(site_key) and is_binary(api_key) do
-    try do
-      formatted_data = Base.url_decode64!(data)
-
-      decoded_data =
-        :aes_cbc128
-        |> :crypto.block_decrypt(generate_key(site_key, api_key), @iv, formatted_data)
-        |> unpad()
-        |> Jason.decode!()
-
-      {:ok, decoded_data}
-    rescue
-      e in ArgumentError -> {:error, e.message}
-      err -> {:error, err}
-    end
-  end
 
-  @spec generate_key(String.t(), String.t()) :: binary()
-  defp generate_key(password, salt) do
-    :crypto.hash(:sha, salt <> password) |> truncate_secret()
-  end
+  alias MultipassEx.DecodingError
+  alias MultipassEx.CryptError
+  alias MultipassEx.Crypt
```

Comment:
mind calling these modules either "Crypto" or "Encryption"? Crypt isn't really a standard term for encryption

Reply:
sure 👍